### DISCUSSION TITLE
LPC persona card not dependend on tertiary text in People search layout

### DISCUSSION THREAD

**Discussion by ivy86** [microsoft-search/pnp-modern-search#3696] [Ideas]
I much appreciate adding the lightweight persona card released recently and the continued effort you are all putting in to it. We require the LPC because we cannot get the other persona card to work in the Teams-application. However, I have noticed that it depends on the UPN being present in the tertiary text field in the Manage People Results settings. I think it also reflected in the code:

`if (this.props.nativeLpc) {
            return <LivePersona upn={processedProps.tertiaryText}`

However, we would like to use this field for something else. Showing the UPN in the people results is not relevant for our users . We already use the fields for other information:
* Image Default value
* Primary field: Default value
* Secondary field: Department
* Tertiary field: Job Title
* Optional field: WorkPhone

I could give up WorkPhone, but it would be great if this functionality could rely on some hidden field.

So, what we'd like:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/57484943/69187f50-5db8-40da-8f4e-6c0eb6178e65)

Versus what we have when using Live Persona Card:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/57484943/8a3c08b9-a30e-485d-98d0-070b8f1c3fb5)

**Comment by wobba** [comment]
Good idea!

**Reply by jblueml** [reply]
I would provide the following extension, if you don't mind:

- additional people field "UPN" for the people layout (no HTML allowed, but Handlebars allowed)
- default value: `{{getUserEmail (slot item @root.slots.UserEmail)}}` (= the same as the "Tertiary text" so far)
- the value of this new field will be passed to `LivePersona`-component in the future (instead of `tertiaryText`)

For backwards compatibility reasons, this field will be added in the `onInit()` function in the `PeopleLayout` class if it is not already present. This should have no effect on the functioning of the WebPart after an update to this new code version.

**Comment by wobba** [comment]
Added with https://github.com/microsoft-search/pnp-modern-search/pull/3835
